compiler/rustc_session/src/config.rs

```diff
@@ -1734,6 +1734,17 @@ fn parse_native_lib_modifiers(
         }
     }
 
+    // Don't allow mixing +bundle with +whole_archive since an rlib may contain multiple
+    // native libs, some of which are +whole_archive and some of which are -whole-archive
+    // and it isn't clear how we can currently handle such a situation correctly.
+    // See https://github.com/rust-lang/rust/issues/88085#issuecomment-901050897
+    if let NativeLibKind::Static { bundle: Some(true), whole_archive: Some(true) } = kind {
+        early_error(
+            error_format,
+            "the linking modifiers +bundle and +whole-archive are not compatible with each other",
+        );
+    }
```

Comment:
- This only covers command line flags, but not attributes.
- `+bundle` is currently the default, so `bundle: None, whole_archive: Some(true)` is also a potential error.
- `+bundle,+whole-archive` is only an issue if we are building a rlib, otherwise it's entirely ok, the `directly_linked.rs` test case uses this configuration, for example.

Reply:
Ah, yeah you're right. All three points should be covered by the latest commit I pushed.